etcdserver/apply.go

```diff
@@ -43,17 +44,18 @@ type applyResult struct {
 	// to being logically reflected by the node. Currently only used for
 	// Compaction requests.
 	physc <-chan struct{}
+	trace *traceutil.Trace
 }
 
 // applierV3 is the interface for processing V3 raft messages
 type applierV3 interface {
 	Apply(r *pb.InternalRaftRequest) *applyResult
 
-	Put(txn mvcc.TxnWrite, p *pb.PutRequest) (*pb.PutResponse, error)
-	Range(txn mvcc.TxnRead, r *pb.RangeRequest) (*pb.RangeResponse, error)
+	Put(txn mvcc.TxnWrite, p *pb.PutRequest) (*pb.PutResponse, *traceutil.Trace, error)
+	Range(ctx context.Context, txn mvcc.TxnRead, r *pb.RangeRequest) (*pb.RangeResponse, error)
```

Comment:
Can we make method signatures consistent for `Put` and `Compaction`?

For me, passing `context.Context` seems cleaner.

Reply:
It is hard to make it consistent with range. For put and compaction request, they need to make raft request first. When etcdserver receives the signal from raft, it will trigger the following apply process.
https://github.com/etcd-io/etcd/blob/c2f23094c875ba1a668192b8324cbc599207bf00/etcdserver/v3_server.go#L113
https://github.com/etcd-io/etcd/blob/c2f23094c875ba1a668192b8324cbc599207bf00/etcdserver/server.go#L1036
I don't think it is a good way to change too many apis inside raft. Therefore, the way I use now is to trace in `apply` then measure the raft request time by calculate the time difference between `apply start time` and `req start time`
Do you think it makes sense?

Reply:
Ok, makes sense.

Btw, we don't really need `ctx` parameter to `Range` unless we want to trace more in auth applier? Can we just create trace object at the top level of `Range` method?

Reply:
Oh, wait. It has to be passed from v3 server. So, nvm.